woocommerce/woocommerce-ios
# Product images UI with CTA to add images (no functionality)

UI part for #1713 

## Changes

- Created a generic `ColumnFlowLayout` collection view flow layout to render a grid layout
- Created `ProductImagesCollectionViewController` to render a collection view of Product images in a grid layout
- Created `ProductImagesViewController` that displays a Call-to-Action for adding images (no-op in this PR) and Product images via a child view controller `ProductImagesCollectionViewController`
- Added action configuration to `ProductFormTableViewDataSource` for adding images `onAddImage`
- In `ProductFormViewController`, implemented `onAddImage` to display `ProductImagesViewController`

## Testing

- Launch the app
- Go to Products tab
- Tap on a simple Product without images
- Tap on the top images header view with "+" action --> a "Photos" screen should be displayed with a CTA to add images
- Navigate back to the Products tab
- Tap on a simple Product with at least one image
- Tap on the "+" cell to the right of the last Product image --> a "Photos" screen should be displayed with a CTA to add images

## Example screenshots

Dark | Light
-- | --
![Simulator Screen Shot - iPhone 8 - 2020-01-09 at 15 26 03](https://user-images.githubusercontent.com/1945542/72047063-acc19c00-32f4-11ea-8151-9dbc1d26729b.png) | ![Simulator Screen Shot - iPhone 8 - 2020-01-09 at 15 26 10](https://user-images.githubusercontent.com/1945542/72047064-ad5a3280-32f4-11ea-9bb3-aeac5bb02c25.png)



Update release notes:
